build-logic/buildquality/src/main/kotlin/gradlebuild.ci-reporting.gradle.kts

```diff
@@ -28,15 +29,18 @@ import gradlebuild.testcleanup.extension.TestFileCleanUpRootExtension
  * Team City.
  */
 
+val testFilesCleanup = extensions.create<TestFileCleanUpExtension>("testFilesCleanup").apply {
+    reportOnly.convention(false)
+}
+
 if (BuildEnvironment.isCiServer && project.name != "gradle-kotlin-dsl-accessors") {
     rootProject.plugins.apply(TestFilesCleanupRootPlugin::class.java)
-    val globalExtension = rootProject.extensions.getByType<TestFileCleanUpRootExtension>()
+    val globalExtension = rootProject.extensions.getByType<TestFilesCleanupBuildServiceRootExtension>()
 
-    val testFilesCleanup = extensions.create<TestFileCleanUpExtension>("testFilesCleanup").apply {
-        reportOnly.convention(false)
-    }
+    val projectExtension = extensions.create<TestFilesCleanupBuildServiceProjectExtension>("testFilesCleanupBuildService")
```

Comment:
Why do you need to create this extension? This only needs to be an object and not an extension. So using `ObjectFactory.newInstance()` should be fine, right?

Reply:
Done.